Gems/LmbrCentral/Code/Source/Shape/BoxShape.cpp

```diff
@@ -267,25 +267,61 @@ namespace LmbrCentral
         }
 
         // transform to world space
-        return m_currentTransform.TransformPoint(AZ::Vector3(x, y, z));
+        AZ::Transform worldTransformWithoutScale = m_currentTransform;
+        const float entityScale = worldTransformWithoutScale.ExtractUniformScale();
+        const AZ::Vector3 scaledTranslationOffset = entityScale * m_currentNonUniformScale * m_boxShapeConfig.m_translationOffset;
+
+        return worldTransformWithoutScale.TransformPoint(AZ::Vector3(x, y, z) + scaledTranslationOffset);
+    }
+
+    AZ::Vector3 BoxShape::GetTranslationOffset()
+    {
+        return m_boxShapeConfig.m_translationOffset;
+    }
+
+    void BoxShape::SetTranslationOffset(const AZ::Vector3& translationOffset)
+    {
+        if (!IsShapeComponentTranslationEnabled())
+        {
+            return;
+        }
+
+        bool shapeChanged = false;
+        {
+            AZStd::unique_lock lock(m_mutex);
+            if (!m_boxShapeConfig.m_translationOffset.IsClose(translationOffset))
+            {
+                m_boxShapeConfig.m_translationOffset = translationOffset;
+                m_intersectionDataCache.InvalidateCache(InvalidateShapeCacheReason::ShapeChange);
+                shapeChanged = true;
+            }
+        }
+
+        if (shapeChanged)
+        {
+            ShapeComponentNotificationsBus::Event(
+                m_entityId, &ShapeComponentNotificationsBus::Events::OnShapeChanged,
+                ShapeComponentNotifications::ShapeChangeReasons::ShapeChanged);
```

Comment:
Crazy semantics question here - should this send "ShapeChanged" or "TransformChanged", since technically it's only the position of the shape that changed?

Reply:
Yeah I debated that for a while. But I was thinking of the offset as part of the shape definition.

Reply:
i can totally buy that. Just wanted to ask the question. :)

Reply:
I'd personally say this is shape data that's changing even if it's in a new place which is why we should prefer `OnShapeChanged` (the entity transform hasn't changed)